python/cuspatial/cuspatial/geometry/geoseries.py

```diff
@@ -0,0 +1,792 @@
+
+    class GeoSeriesILocIndexer:
+        """
+        Each row of a GeoSeries is one of the six types: Point, MultiPoint,
+        LineString, MultiLineString, Polygon, or MultiPolygon.
+        """
+
+        def __init__(self, sr):
+            self._sr = sr
+
+        def __getitem__(self, index):
+            if not isinstance(index, slice):
+                return self._getitem_int(index)
+            else:
+                raise NotImplementedError
+                # This slice functionality is not ready yet
+                # return self._getitem_slice(index)
+
+        def _getitem_int(self, index):
+            type_map = {
+                "p": gpuPoint,
+                "mp": gpuMultiPoint,
+                "l": gpuLineString,
+                "ml": gpuMultiLineString,
+                "poly": gpuPolygon,
+                "mpoly": gpuMultiPolygon,
+            }
+            return type_map[self._sr.types[index]](self._sr, index)
+
+
+class GpuCoordinateArray:
+    def __init__(self, xy, z=None):
+        """
+        A GeoArrow column of points. The GpuCoordinateArray stores all of the
+        points within a single data source, typically a cuspatial.GeoSeries,
+        in the format specified by GeoArrow.
+        """
+        self._xy = xy
+        self._z = z
+
+    @property
+    def xy(self):
+        """
+        The coordinates of this Geometry Column in interleaved format
+        [x,y,...,x,y].
+        """
+        return self._xy
+
+    @xy.setter
+    def xy(self, xy):
+        self._xy = xy
+
+    @property
+    def z(self):
+        """
+        An optional third dimension for this Geometry.
+        """
+        return self._z
+
+    @z.setter
+    def z(self, z):
+        self._z = z
+
+    def __getitem__(self, index):
+        if isinstance(index, slice):
+            new_index = slice(index.start * 2, index.stop * 2 + 1, index.step)
+            return self.xy.iloc[new_index]
+        return self.xy.iloc[(index * 2) : (index * 2) + 2]
+
+    def __repr__(self):
+        return f"xy:\n" f"{self.xy.__repr__()}\n"
+
+    def copy(self, deep=True):
+        """
+        Create a copy of all points.
+
+        Parameters
+        ----------
+        deep : Boolean.
+            If set to False, a new object referencing the same
+            set of GPU memory will be created. If True, GPU memory will be
+            copied.
+        """
+        if hasattr(self, "_z"):
+            z = self.z.copy(deep)
+        else:
+            z = None
+        result = GpuCoordinateArray(self.xy.copy(deep), z)
+        return result
+
+    @property
+    def x(self):
+        """
+        Return packed x-coordinates of this GpuGeometryArray object.
+        """
+        return self.xy[slice(0, None, 2)].reset_index(drop=True)
+
+    @property
+    def y(self):
+        """
+        Return packed y-coordinates of this GpuGeometryArray object.
+        """
+        return self.xy[slice(1, None, 2)].reset_index(drop=True)
```

Comment:
The fact that we have to drop an index for this operation points to something being up with these implementations. Can we not gather the underlying columns and then assign a RangeIndex when creating the Series around it?

Reply:
The original data, `.xy` uses a default index (I assume a RangeIndex?). When I slice out the odd values here with `slice(1, None, 2)` I also slice out the odd index values, so the resulting index is `[1, 3, 5...]` when it is expected to be `[0, 1, 2...]`. I `reset_index` to go back to the default index. Isn't this the equivalent of doing the following?

    return cudf.Series(self.xy[slice(1, None, 2)], index=cudf.RangeIndex(0, len(self.xy) / 2))